src/environmentd/src/http.rs

```diff
@@ -303,9 +315,41 @@ where
                 "adapter client missing".into(),
             )
         })?;
-        AuthedClient::new(&adapter_client, user.clone())
+        let mut client = AuthedClient::new(&adapter_client, user.clone())
             .await
-            .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
+            .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))?;
+
+        // Apply options that were provided in query params.
+        let session = client.0.session();
+        let maybe_options = if params.options.is_empty() {
+            // It's possible 'options' simply wasn't provided, we don't want that to
+            // count as a failure to deserialize
+            Ok(BTreeMap::<String, String>::default())
+        } else {
+            serde_json::from_str(&params.options)
+        };
+
+        if let Ok(options) = maybe_options {
+            for (key, val) in options {
+                const LOCAL: bool = false;
+                if let Err(err) = session.vars_mut().set(&key, VarInput::Flat(&val), LOCAL) {
+                    session.add_notice(AdapterNotice::BadStartupSetting {
+                        name: key.to_string(),
+                        reason: err.to_string(),
+                    })
+                }
+            }
+        } else {
+            // If we fail to deserialize options, emit a Notice.
```

Comment:
This should error and close the connection, not just send a notice.

Reply:
Updated!